src/fs/file.rs

```diff
@@ -171,6 +171,89 @@ impl File {
         op.read().await
     }
 
+    /// Read the exact number of bytes required to fill `buf` at the specified
+    /// offset from the file.
+    ///
+    /// This function reads as many as bytes as necessary to completely fill the
+    /// specified buffer `buf`.
+    ///
+    /// # Return
+    ///
+    /// The method returns the operation result and the same buffer value passed
+    /// as an argument.
+    ///
+    /// If the method returns [`Ok(())`], then the read was successful.
+    ///
+    /// # Errors
+    ///
+    /// If this function encounters an error of the kind [`ErrorKind::Interrupted`]
+    /// then the error is ignored and the operation will continue.
+    ///
+    /// If this function encounters an "end of file" before completely filling
+    /// the buffer, it returns an error of the kind [`ErrorKind::UnexpectedEof`].
+    /// The buffer is returned on error.
+    ///
+    /// If this function encounters any form of I/O or other error, an error
+    /// variant will be returned. The buffer is returned on error.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use tokio_uring::fs::File;
+    ///
+    /// fn main() -> Result<(), Box<dyn std::error::Error>> {
+    ///     tokio_uring::start(async {
+    ///         let f = File::open("foo.txt").await?;
+    ///         let buffer = Vec::with_capacity(10);
+    ///
+    ///         // Read up to 10 bytes
+    ///         let (res, buffer) = f.read_exact_at(buffer, 0).await;
+    ///         res?;
+    ///
+    ///         println!("The bytes: {:?}", buffer);
+    ///
+    ///         // Close the file
+    ///         f.close().await?;
+    ///         Ok(())
+    ///     })
+    /// }
+    /// ```
+    ///
+    /// [`ErrorKind::Interrupted`]: std::io::ErrorKind::Interrupted
+    /// [`ErrorKind::UnexpectedEof`]: std::io::ErrorKind::UnexpectedEof
+    pub async fn read_exact_at<T: IoBufMut>(
+        &self,
+        mut buf: T,
+        pos: u64,
+    ) -> crate::BufResult<(), T> {
+        let buf_len = buf.bytes_total();
```

Comment:
Should we check for wrapping around of `buf_len + pos` here?

Reply:
Not sure about this, is there any similar checks in other functions?

Reply:
Let's resolve this unless someone has an example in std or tokio where such a wrap around check is made?